content/Registration and Authentication in your Django app with dj-rest-auth part1- Bruno Michetti.md

```diff
@@ -0,0 +1,214 @@
+
+```python
+# serializers.py in the users Django app
+from django.db import transaction
+from rest_framework import serializers
+from dj_rest_auth.registration.serializers import RegisterSerializer
+
+from users.models import GENDER_SELECTION
+
+
+class CustomRegisterSerializer(RegisterSerializer):
+    gender = serializers.ChoiceField(choices=GENDER_SELECTION)
+    phone_number = serializers.CharField(max_length=30)
+
+    # Define transaction.atomic to rollback the save operation in case of error
+    @transaction.atomic
+    def save(self, request):
+        user = super().save(request)
+        user.gender = self.data.get('gender')
+        user.phone_number = self.data.get('phone_number')
+        user.save()
+        return user
+
+```
+
+Now we can't forget to tell dj-rest-auth to use the recently created serializer. Let's add the `CustomRegisterSerializer` to the settings:
+
+```python
+REST_AUTH_REGISTER_SERIALIZERS = {
+    'REGISTER_SERIALIZER': 'users.serializers.CustomRegisterSerializer',
+}
+```
+
+After this, if we go to the registration endpoint, we would see something like this:
+
+<img src="images/dj-rest-auth-registration-example.png" alt="Registration endpoint" width="900"/>
+
+The username appears in the browsable endpoint, but thanks to the configuration you can leave it empty.  
+And thats it! we have added a customized registration endpoint to our Django app. Now we can create users with email, gender and phone number, and test the login too.
+
+### Use of JWT
+
+When an app manages authentication, a good and standard way is using [JWT (JSON web tokens)](https://auth0.com/docs/tokens/json-web-tokens). JWTs are a good way of securely transmitting information between parties because they can be signed, which means you can be sure that the senders are who they say they are. We can easily configure our Django app to use JWT. First we need to install `djangorestframework-simplejwt`:
+
+```python
+pip install djangorestframework-simplejwt
+```
+
+Then add to the settings:
+
+```python
+REST_FRAMEWORK = {
+    'DEFAULT_AUTHENTICATION_CLASSES': (
+        'dj_rest_auth.jwt_auth.JWTCookieAuthentication',
+    ),
+}
+
+REST_USE_JWT = True
+JWT_AUTH_COOKIE = 'my-app-auth' # The cookie key name can be the one you want
+```
+
+And we are good to go! We have an app that manages the authentication with JWT.
+
+#### Note:
+
+To do all of the mentioned things on previous examples, first you need to follow the dj-rest-auth installation steps, also you need to have installed django allauth, and finally make sure you have the authentication backends in your settings:
+
+```python
+AUTHENTICATION_BACKENDS = [
+    # allauth specific authentication methods, such as login by e-mail
+    'allauth.account.auth_backends.AuthenticationBackend',
+    # Needed to login by username in Django admin, regardless of allauth
+    'django.contrib.auth.backends.ModelBackend',
+]
+```
```

Comment:
Maybe start with this the section and then explain how to integrate JWT lib, in this way is easier for the readers to follow the steps

Reply:
Where do you think to put this note?

Reply:
I was thinking move this note (not as a note) at the begging of the `Configuration` chapter, indicated at least the documentation links where are the instructions for installing django allauth